omegaconf/omegaconf.py

```diff
@@ -829,6 +829,14 @@ def to_yaml(cfg: Any, *, resolve: bool = False, sort_keys: bool = False) -> str:
         :param sort_keys: If True, will print dict keys in sorted order. default False.
         :return: A string containing the yaml representation.
         """
+        if not (
+            is_primitive_container(cfg)
+            or is_structured_config(cfg)
+            or OmegaConf.is_config(cfg)
+        ):
+            raise ValueError(
+                "to_yaml expects input of type DictConfig / ListConfig / dict / list / StructuredConfig."
+            )
```

Comment:
1. Move the check to ensure_container.
2. Error should be something like:
```
            raise ValueError(
                "Invalid input. Supported one of [dict,list,DictConfig,ListConfig,dataclass,dataclass instance, attr class, attr class instance]"
            )
```

Reply:
Done in https://github.com/omry/omegaconf/pull/568/commits/9336a17f0d4be1acd2bf178c3f4f1f14707730b9.